### PR TITLE
Added logger to imp

### PR DESCRIPTION
**Pull request by glanvl**
Adapted the logger to use the new spp-logger function/library.

### CODE REVIEW

File: add_regionless_method.py [hunk 1]
```diff
@@ -14,9 +14,11 @@ def handle_error(self, e, data, **kwargs):
         raise ValueError(f"Error validating runtime params: {e}")
 
     bpm_queue_url = fields.Str(required=True)
+    environment = fields.Str(required=True)
     data = fields.List(fields.Dict, required=True)
```
Comment (Piwington) [inline add_regionless_method.py:18]: D before E.

File: add_regionless_wrangler.py [hunk 2]
```diff
@@ -103,11 +117,13 @@ def lambda_handler(event, context):
         payload = {
             "RuntimeVariables": {
                 "bpm_queue_url": bpm_queue_url,
+                "environment": environment,
                 "data": json.loads(
```
Comment (Piwington) [inline add_regionless_wrangler.py:121]: D before E.

File: apply_factors_method.py [hunk 3]
```diff
@@ -19,8 +19,10 @@ def handle_error(self, e, data, **kwargs):
         raise ValueError(f"Error validating runtime params: {e}")
 
     bpm_queue_url = fields.Str(required=True)
+    environment = fields.Str(required=True)
     data = fields.List(fields.Dict, required=True)
```
Comment (Piwington) [inline apply_factors_method.py:23]: D before E.

File: apply_factors_method.py [hunk 4]
```diff
@@ -19,8 +19,10 @@ def handle_error(self, e, data, **kwargs):
...
     questions_list = fields.List(fields.String, required=True)
+    survey = fields.Str(required=True)
     sum_columns = fields.Nested(SumSchema, many=True, required=True)
```
Comment (Piwington) [inline apply_factors_method.py:26]: M before R.

File: apply_factors_method.py [hunk 5]
```diff
@@ -42,22 +43,35 @@ def lambda_handler(event, context):
...
 
-        logger.info("Validated parameters.")
-
         # Runtime Variables
         bpm_queue_url = runtime_variables["bpm_queue_url"]
+        environment = runtime_variables["environment"]
         json_data = runtime_variables["data"]
         questions_list = runtime_variables["questions_list"]
+        survey = runtime_variables["survey"]
         sum_columns = runtime_variables["sum_columns"]
```
Comment (Piwington) [inline apply_factors_method.py:58]: M before R.

File: apply_factors_wrangler.py [hunk 6]
```diff
@@ -43,6 +43,7 @@ def handle_error(self, e, data, **kwargs):
 
     bpm_queue_url = fields.Str(required=True)
     current_data = fields.Str(required=True)
+    environment = fields.Str(required=True)
     distinct_values = fields.List(fields.String, required=True)
```
Comment (Piwington) [inline apply_factors_wrangler.py:47]: D before E.

File: apply_factors_wrangler.py [hunk 7]
```diff
@@ -53,6 +54,7 @@ def handle_error(self, e, data, **kwargs):
     questions_list = fields.List(fields.String, required=True)
     unique_identifier = fields.List(fields.String, required=True)
     sns_topic_arn = fields.Str(required=True)
+    survey = fields.Str(required=True)
     sum_columns = fields.List(fields.Dict, required=True)
```
Comment (Piwington) [inline apply_factors_wrangler.py:58]: M before R.

File: apply_factors_wrangler.py [hunk 8]
```diff
@@ -99,6 +97,7 @@ def lambda_handler(event, context):
 
         # Runtime Variables
         bpm_queue_url = runtime_variables["bpm_queue_url"]
+        environment = runtime_variables["environment"]
         current_data = runtime_variables["current_data"]
         distinct_values = runtime_variables["distinct_values"]
```
Comment (Piwington) [inline apply_factors_wrangler.py:102]: C and D before E.

File: apply_factors_wrangler.py [hunk 9]
```diff
@@ -112,8 +111,23 @@ def lambda_handler(event, context):
         sns_topic_arn = runtime_variables["sns_topic_arn"]
         sum_columns = runtime_variables["sum_columns"]
         total_steps = runtime_variables["total_steps"]
+        survey = runtime_variables["survey"]
```
Comment (Piwington) [inline apply_factors_wrangler.py:114]: S before T.

File: apply_factors_wrangler.py [hunk 10]
```diff
@@ -209,11 +223,13 @@ def lambda_handler(event, context):
         payload = {
             "RuntimeVariables": {
                 "bpm_queue_url": bpm_queue_url,
+                "environment": environment,
                 "data": json.loads(
```
Comment (Piwington) [inline apply_factors_wrangler.py:227]: D before E.

File: apply_factors_wrangler.py [hunk 11]
```diff
@@ -209,11 +223,13 @@ def lambda_handler(event, context):
...
                     non_responders_with_factors.to_json(orient="records")),
                 "questions_list": questions_list,
                 "sum_columns": sum_columns,
-                "run_id": run_id
+                "run_id": run_id,
```
Comment (Piwington) [inline apply_factors_wrangler.py:231]: Also noticed that this R should be before S.

File: atypicals_method.py [hunk 12]
```diff
@@ -17,8 +17,10 @@ def handle_error(self, e, data, **kwargs):
         raise ValueError(f"Error validating runtime params: {e}")
 
     bpm_queue_url = fields.Str(required=True)
+    environment = fields.Str(required=True)
     data = fields.List(fields.Dict, required=True)
```
Comment (Piwington) [inline atypicals_method.py:21]: D before E.

File: atypicals_wrangler.py [hunk 13]
```diff
@@ -102,9 +115,11 @@ def lambda_handler(event, context):
         payload = {
             "RuntimeVariables": {
                 "bpm_queue_url": bpm_queue_url,
+                "environment": environment,
                 "data": json.loads(data_json),
```
Comment (Piwington) [inline atypicals_wrangler.py:119]: D before E.

File: calculate_imputation_factors_method.py [hunk 14]
```diff
@@ -16,10 +16,12 @@ def handle_error(self, e, data, **kwargs):
         raise ValueError(f"Error validating runtime params: {e}")
 
     bpm_queue_url = fields.Str(required=True)
+    environment = fields.Str(required=True)
     data = fields.List(fields.Dict, required=True)
```
Comment (Piwington) [inline calculate_imputation_factors_method.py:20]: D before E.

File: calculate_imputation_factors_method.py [hunk 15]
```diff
@@ -56,15 +56,29 @@ def lambda_handler(event, context):
 
         factors = factors_schema().load(factors_parameters)
 
-        logger.info("Validated parameters.")
-
         # Runtime Variables
         bpm_queue_url = runtime_variables["bpm_queue_url"]
+        environment = runtime_variables["environment"]
         df = pd.DataFrame(runtime_variables["data"])
         distinct_values = runtime_variables["distinct_values"]
```
Comment (Piwington) [inline calculate_imputation_factors_method.py:63]: D before E.

File: calculate_imputation_factors_wrangler.py [hunk 16]
```diff
@@ -32,13 +32,15 @@ def handle_error(self, e, data, **kwargs):
         raise ValueError(f"Error validating runtime params: {e}")
 
     bpm_queue_url = fields.Str(required=True)
+    environment = fields.Str(required=True)
     distinct_values = fields.List(fields.String, required=True)
```
Comment (Piwington) [inline calculate_imputation_factors_wrangler.py:36]: D before E.

File: calculate_imputation_factors_wrangler.py [hunk 17]
```diff
@@ -73,24 +72,38 @@ def lambda_handler(event, context):
...
-
         # Environment Variables
         bucket_name = environment_variables["bucket_name"]
         method_name = environment_variables["method_name"]
         run_environment = environment_variables["run_environment"]
 
         # Runtime Variables
         bpm_queue_url = runtime_variables["bpm_queue_url"]
+        environment = runtime_variables["environment"]
         distinct_values = runtime_variables["distinct_values"]
```
Comment (Piwington) [inline calculate_imputation_factors_wrangler.py:83]: D before E.

File: calculate_imputation_factors_wrangler.py [hunk 18]
```diff
@@ -106,11 +119,13 @@ def lambda_handler(event, context):
         payload = {
             "RuntimeVariables": {
                 "bpm_queue_url": bpm_queue_url,
+                "environment": environment,
                 "data": json.loads(data.to_json(orient="records")),
```
Comment (Piwington) [inline calculate_imputation_factors_wrangler.py:123]: D before E.

File: calculate_means_method.py [hunk 19]
```diff
@@ -16,9 +16,11 @@ def handle_error(self, e, data, **kwargs):
         raise ValueError(f"Error validating runtime params: {e}")
 
     bpm_queue_url = fields.Str(required=True)
+    environment = fields.Str(required=True)
     data = fields.List(fields.Dict, required=True)
```
Comment (Piwington) [inline calculate_means_method.py:20]: D before E.

File: calculate_means_method.py [hunk 20]
```diff
@@ -42,22 +43,35 @@ def lambda_handler(event, context):
...
 
         runtime_variables = RuntimeSchema().load(event["RuntimeVariables"])
 
-        logger.info("Validated parameters.")
-
         # Runtime Variables
         bpm_queue_url = runtime_variables["bpm_queue_url"]
+        environment = runtime_variables["environment"]
         json_data = runtime_variables["data"]
         distinct_values = runtime_variables["distinct_values"]
```
Comment (Piwington) [inline calculate_means_method.py:56]: Not your fault but while you are editing D before J and E.

File: calculate_means_wrangler.py [hunk 21]
```diff
@@ -31,11 +31,13 @@ def handle_error(self, e, data, **kwargs):
         raise ValueError(f"Error validating runtime params: {e}")
 
     bpm_queue_url = fields.Str(required=True)
+    environment = fields.Str(required=True)
     distinct_values = fields.List(fields.String, required=True)
```
Comment (Piwington) [inline calculate_means_wrangler.py:35]: D before E.

File: calculate_means_wrangler.py [hunk 22]
```diff
@@ -71,22 +69,36 @@ def lambda_handler(event, context):
...
-
         # Environment Variables
         bucket_name = environment_variables["bucket_name"]
         method_name = environment_variables["method_name"]
         run_environment = environment_variables["run_environment"]
 
         # Runtime Variables
         bpm_queue_url = runtime_variables["bpm_queue_url"]
+        environment = runtime_variables["environment"]
         distinct_values = runtime_variables["distinct_values"]
```
Comment (Piwington) [inline calculate_means_wrangler.py:80]: D before E.

File: calculate_means_wrangler.py [hunk 23]
```diff
@@ -105,10 +117,12 @@ def lambda_handler(event, context):
         payload = {
             "RuntimeVariables": {
                 "bpm_queue_url": bpm_queue_url,
+                "environment": environment,
                 "data": json.loads(data_json),
                 "distinct_values": distinct_values,
```
Comment (Piwington) [inline calculate_means_wrangler.py:122]: D before E.

File: calculate_movement_method.py [hunk 24]
```diff
@@ -16,12 +16,14 @@ def handle_error(self, e, data, **kwargs):
         raise ValueError(f"Error validating runtime params: {e}")
 
     bpm_queue_url = fields.Str(required=True)
+    environment = fields.Str(required=True)
     movement_type = fields.Str(required=True)
     data = fields.List(fields.Dict, required=True)
     questions_list = fields.List(fields.String, required=True)
     current_period = fields.Str(required=True)
```
Comment (Piwington) [inline calculate_movement_method.py:23]: Another not your fault but pretty please. C, Q, D not in the right places here.

File: calculate_movement_method.py [hunk 25]
```diff
@@ -51,18 +52,32 @@ def lambda_handler(event, context):
...
 
-        logger.info("Validated parameters.")
-
         # Runtime Variables
         bpm_queue_url = runtime_variables["bpm_queue_url"]
+        environment = runtime_variables["environment"]
         movement_type = runtime_variables["movement_type"]
         json_data = runtime_variables["data"]
         questions_list = runtime_variables["questions_list"]
         current_period = runtime_variables["current_period"]
```
Comment (Piwington) [inline calculate_movement_method.py:61]: As about but J not D.

File: calculate_movement_wrangler.py [hunk 26]
```diff
@@ -30,6 +30,7 @@ def handle_error(self, e, data, **kwargs):
         raise ValueError(f"Error validating runtime params: {e}")
 
     bpm_queue_url = fields.Str(required=True)
+    environment = fields.Str(required=True)
     current_data = fields.Str(required=True)
```
Comment (Piwington) [inline calculate_movement_wrangler.py:34]: C before E.

File: calculate_movement_wrangler.py [hunk 27]
```diff
@@ -41,6 +42,7 @@ def handle_error(self, e, data, **kwargs):
     previous_data = fields.Str(required=True)
     questions_list = fields.List(fields.String, required=True)
     sns_topic_arn = fields.Str(required=True)
+    survey = fields.Str(required=True)
     unique_identifier = fields.List(fields.String, required=True)
     total_steps = fields.Str(required=True)
```
Comment (Piwington) [inline calculate_movement_wrangler.py:47]: Not your fault but T before U.

File: calculate_movement_wrangler.py [hunk 28]
```diff
@@ -68,28 +69,25 @@ def lambda_handler(event, context):
...
-
         # Environment Variables
         bucket_name = environment_variables["bucket_name"]
         method_name = environment_variables["method_name"]
         response_type = environment_variables["response_type"]
 
         # Runtime Variables
         bpm_queue_url = runtime_variables["bpm_queue_url"]
+        environment = runtime_variables["environment"]
         current_data = runtime_variables["current_data"]
```
Comment (Piwington) [inline calculate_movement_wrangler.py:91]: C before E.

File: calculate_movement_wrangler.py [hunk 29]
```diff
@@ -170,13 +183,15 @@ def lambda_handler(event, context):
             json_payload = {
                 "RuntimeVariables": {
                     "bpm_queue_url": bpm_queue_url,
+                    "environment": environment,
                     "data": json.loads(json_ordered_data),
```
Comment (Piwington) [inline calculate_movement_wrangler.py:187]: D before E.

File: calculate_movement_wrangler.py [hunk 30]
```diff
@@ -170,13 +183,15 @@ def lambda_handler(event, context):
...
                     "movement_type": movement_type,
                     "questions_list": questions_list,
                     "current_period": period,
```
Comment (Piwington) [inline calculate_movement_wrangler.py:190]: Not you but C and Q.

File: iqrs_method.py [hunk 31]
```diff
@@ -16,9 +16,11 @@ def handle_error(self, e, data, **kwargs):
         raise ValueError(f"Error validating runtime params: {e}")
 
     bpm_queue_url = fields.Str(required=True)
+    environment = fields.Str(required=True)
     data = fields.List(fields.Dict, required=True)
     distinct_values = fields.List(fields.String, required=True)
```
Comment (Piwington) [inline iqrs_method.py:21]: D before E.

File: iqrs_method.py [hunk 32]
```diff
@@ -40,23 +41,36 @@ def lambda_handler(event, context):
...
         run_id = event["RuntimeVariables"]["run_id"]
 
         runtime_variables = RuntimeSchema().load(event["RuntimeVariables"])
 
-        logger.info("Validated parameters.")
-
         # Runtime Variables
         bpm_queue_url = runtime_variables["bpm_queue_url"]
+        environment = runtime_variables["environment"]
         distinct_values = runtime_variables["distinct_values"]
```
Comment (Piwington) [inline iqrs_method.py:53]: D before E.

File: iqrs_wrangler.py [hunk 33]
```diff
@@ -31,11 +31,13 @@ def handle_error(self, e, data, **kwargs):
         raise ValueError(f"Error validating runtime params: {e}")
 
     bpm_queue_url = fields.Str(required=True)
+    environment = fields.Str(required=True)
     distinct_values = fields.List(fields.String, required=True)
```
Comment (Piwington) [inline iqrs_wrangler.py:35]: D before E.

File: iqrs_wrangler.py [hunk 34]
```diff
@@ -69,22 +68,37 @@ def lambda_handler(event, context):
...
-
         # Environment Variables
         bucket_name = environment_variables["bucket_name"]
         method_name = environment_variables["method_name"]
         run_environment = environment_variables["run_environment"]
 
         # Runtime Variables
         bpm_queue_url = runtime_variables["bpm_queue_url"]
+        environment = runtime_variables["environment"]
         distinct_values = runtime_variables["distinct_values"]
```
Comment (Piwington) [inline iqrs_wrangler.py:79]: D before E.

File: iqrs_wrangler.py [hunk 35]
```diff
@@ -101,10 +115,12 @@ def lambda_handler(event, context):
         payload = {
             "RuntimeVariables": {
                 "bpm_queue_url": bpm_queue_url,
+                "environment": environment,
                 "data": json.loads(data_json),
                 "distinct_values": distinct_values,
```
Comment (Piwington) [inline iqrs_wrangler.py:120]: D before E.

File: recalculate_means_wrangler.py [hunk 36]
```diff
@@ -29,11 +29,13 @@ def handle_error(self, e, data, **kwargs):
         raise ValueError(f"Error validating runtime params: {e}")
 
     bpm_queue_url = fields.Str(required=True)
+    environment = fields.Str(required=True)
     distinct_values = fields.List(fields.String, required=True)
```
Comment (Piwington) [inline recalculate_means_wrangler.py:33]: D before E.

File: recalculate_means_wrangler.py [hunk 37]
```diff
@@ -70,22 +70,36 @@ def lambda_handler(event, context):
...
-
         # Environment Variables
         bucket_name = environment_variables["bucket_name"]
         method_name = environment_variables["method_name"]
         run_environment = environment_variables["run_environment"]
 
         # Runtime Variables
         bpm_queue_url = runtime_variables["bpm_queue_url"]
+        environment = runtime_variables["environment"]
         distinct_values = runtime_variables["distinct_values"]
```
Comment (Piwington) [inline recalculate_means_wrangler.py:81]: D before E.

File: recalculate_means_wrangler.py [hunk 38]
```diff
@@ -103,10 +117,12 @@ def lambda_handler(event, context):
         payload = {
             "RuntimeVariables": {
                 "bpm_queue_url": bpm_queue_url,
+                "environment": environment,
                 "data": json.loads(data_json),
                 "distinct_values": distinct_values,
```
Comment (Piwington) [inline recalculate_means_wrangler.py:122]: D before E.

File: tests/test_imputation.py [hunk 39]
```diff
@@ -264,19 +285,22 @@
               }
             }
           ],
-        "unique_identifier": ["responder_id"]
+        "unique_identifier": ["responder_id"],
+        "survey": "bmi_sg"
```
Comment (Piwington) [inline tests/test_imputation.py:289]: S before U.

File: tests/test_imputation.py [hunk 40]
```diff
@@ -264,19 +285,22 @@
...
     }
 }
 
 wrangler_atypicals_runtime_variables = {
     "RuntimeVariables": {
         "bpm_queue_url": "fake_bpm_queue_url",
         "total_steps": "4",
```
Comment (Piwington) [inline tests/test_imputation.py:296]: Not you but T should be lower.

File: tests/test_imputation.py [hunk 41]
```diff
@@ -346,18 +378,21 @@
...
         "run_id": "bob",
         "sns_topic_arn": "fake_sns_arn",
+        "survey": "bmi_sg"
     }
 }
 
 wrangler_regionless_runtime_variables = {
     "RuntimeVariables": {
         "bpm_queue_url": "fake_bpm_queue_url",
         "total_steps": "4",
```
Comment (Piwington) [inline tests/test_imputation.py:394]: And not you but T.

**Review by Piwington — CHANGES_REQUESTED** [review]
All changes are Alphabetical. Actual code is looking find to me.

**Review by Piwington — APPROVED** [review]
Perfect. 👍

**Review by rhodriguerrier — APPROVED** [review]

**Pull request merged**